cmd/lotus-shed/ledger.go

```diff
@@ -27,14 +31,37 @@ var ledgerCmd = &cli.Command{
 
 var ledgerListAddressesCmd = &cli.Command{
 	Name: "list",
+	Flags: []cli.Flag{
+		&cli.BoolFlag{
+			Name: "print-balances",
+			Usage: "print balances",
+			Aliases: []string{"b"},
+		},
+	},
 	Action: func(cctx *cli.Context) error {
+		var api api.FullNode
+		if cctx.Bool("print-balances") {
+			a, closer, err := lcli.GetFullNodeAPI(cctx)
+			if err != nil {
+				return err
+			}
+
+			api = a
+
+			defer closer()
+		}
+		ctx := lcli.ReqContext(cctx)
 
 		fl, err := ledgerfil.FindLedgerFilecoinApp()
 		if err != nil {
 			return err
 		}
 
 		for i := 0; i < 20; i++ {
```

Comment:
When iterating while looking at the balance, we should iterate 20 keys in front of the last account with balance. (BIP44)